presto-iceberg/src/main/java/com/facebook/presto/iceberg/procedure/SetTablePropertyProcedure.java

```diff
@@ -1,98 +0,0 @@
-import com.facebook.presto.iceberg.IcebergUtil;
-import com.facebook.presto.spi.ConnectorSession;
-import com.facebook.presto.spi.SchemaTableName;
-import com.facebook.presto.spi.connector.ConnectorMetadata;
-import com.facebook.presto.spi.procedure.Procedure;
-import com.google.common.collect.ImmutableList;
-import org.apache.iceberg.Table;
-
-import javax.inject.Inject;
-import javax.inject.Provider;
-
-import java.lang.invoke.MethodHandle;
-
-import static com.facebook.presto.common.block.MethodHandleUtil.methodHandle;
-import static com.facebook.presto.common.type.StandardTypes.VARCHAR;
-import static java.util.Objects.requireNonNull;
-
-public class SetTablePropertyProcedure
-        implements Provider<Procedure>
-{
-    private static final Logger LOG = Logger.get(SetTablePropertyProcedure.class);
-    private static final MethodHandle SET_TABLE_PROPERTY = methodHandle(
-            SetTablePropertyProcedure.class,
-            "setTableProperty",
-            ConnectorSession.class,
-            String.class,
-            String.class,
-            String.class,
-            String.class);
-
-    private final IcebergConfig config;
-    private final IcebergMetadataFactory metadataFactory;
-    private final HdfsEnvironment hdfsEnvironment;
-
-    @Inject
-    public SetTablePropertyProcedure(
-            IcebergConfig config,
-            IcebergMetadataFactory metadataFactory,
-            HdfsEnvironment hdfsEnvironment)
-    {
-        this.config = requireNonNull(config);
-        this.metadataFactory = requireNonNull(metadataFactory);
-        this.hdfsEnvironment = requireNonNull(hdfsEnvironment);
-    }
-
-    @Override
-    public Procedure get()
-    {
-        return new Procedure(
-                "system",
-                "set_table_property",
-                ImmutableList.of(
-                        new Procedure.Argument("schema", VARCHAR),
-                        new Procedure.Argument("table", VARCHAR),
-                        new Procedure.Argument("key", VARCHAR),
-                        new Procedure.Argument("value", VARCHAR)),
-                SET_TABLE_PROPERTY.bindTo(this));
-    }
-
-    /**
-     * Sets a key-value string pair inside an Iceberg table's properties
-     *
-     * @param session the client's session information
-     * @param schema the schema where the table exists
-     * @param table the name of the table to sample from
-     */
-    public void setTableProperty(ConnectorSession session, String schema, String table, String key, String value)
-    {
-        ConnectorMetadata metadata = metadataFactory.create();
-        IcebergTableName tableName = IcebergTableName.from(table);
-        SchemaTableName schemaTableName = new SchemaTableName(schema, tableName.getTableName());
-        Table icebergTable = IcebergUtil.getIcebergTable(metadata, session, schemaTableName);
-
-        icebergTable.updateProperties()
-                .set(key, value)
-                .commit();
-    }
-}
```

Comment:
Is there a good reason to remove it? If I recall, the reason I added this was because iceberg's table properties are generic string-string key-value pairs. Presto doesn't allow setting any arbitrary properties through regular SQL syntax though. So this adds some useful functionality for users who might have some arbitrary table-level metadata to store.

Reply:
> Besides, this PR removes a procedure set_table_property which is completely unused.

@hantangwangd It was unused as it's missing binding in IcebergCommonModule, so the Procedure wasn't even enabled till now. 

We can add the binding and enable it as it should be useful as Zac mentioned. What do you think?

Reply:
I just saw that this procedure was not used at all, not even registered as @imjalpreet mentioned. I do not have a strong feeling to remove it, since the PR #21495 has been blocked for quite some time, I would like to register and enable this procedure if you think it's necessary. Let me know your thoughts @ZacBlanco @imjalpreet.

Reply:
Now keep `SetTablePropertyProcedure` as it is, maybe it's better to handle it in a later dedicated PR.

Reply:
Sounds good, we can enable the procedure and add some tests in a follow-up PR.